src/generators.rs

```diff
@@ -12,14 +12,243 @@ use blockstack_lib::util::hash::to_hex;
+        1 => Value::Int(rng.gen()),
+        2 => Value::Optional(OptionalData{ data: Some(Box::new(Value::Bool(rng.gen_bool(0.5))))}),
+        n => {
+            let mult = n / 8;
+            // assume n is a multiple of 8
+            let mut type_map = BTreeMap::new();
+            let mut data_map = BTreeMap::new();
+            let value_type_sig = TypeSignature::IntType;
+            for i in 0..mult {
+                let id_name = if i < 10 { format!("id--{}", i) } else { format!("id-{}", i) };
+                let clarity_name = ClarityName::try_from(id_name).unwrap();
+                let value = Value::Int(rng.gen());
+                type_map.insert(clarity_name.clone(), value_type_sig.clone());
+                data_map.insert(clarity_name, value);
+            }
+            let tuple_data = TupleData{
+                type_signature: TupleTypeSignature::try_from(type_map).unwrap(),
+                data_map,
+            };
+
+           Value::Tuple(tuple_data)
+        }
+    }
+}
+
+// make sized type sigs for AnalysisTypeCheck
+fn make_sized_type_sig(input_size: u16) -> TypeSignature {
+    let mut rng = rand::thread_rng();
+    match input_size {
+        1 => IntType,
+        2 => TypeSignature::OptionalType(Box::new(TypeSignature::BoolType)),
+        n => {
+            // assume n is a multiple of 8
+            let type_list = [IntType];
+            let mut type_map = Vec::new();
+            let mult = n / 8;
+            for i in 0..mult {
+                let id_name = if i < 10 { "id--" } else { "id-" };
+                let name = ClarityName::try_from(format!("{}{}", id_name, i)).unwrap();
+                let type_sig = type_list.choose(&mut rng).unwrap().clone();
+                type_map.push((name, type_sig));
+            }
+            TupleType(TupleTypeSignature::try_from(type_map).unwrap())
+        }
+    }
+}
+
+pub fn make_sized_type_sig_map(input_sizes: Vec<u16>) -> HashMap<u16, TypeSignature> {
+    let mut ret_map = HashMap::new();
+    for i in input_sizes {
+        let val = make_sized_type_sig(i);
+        ret_map.insert(i, val);
+    }
+    ret_map
+}
+
+pub fn helper_make_sized_clarity_value(input_size: u16) -> String {
+    let mut rng = rand::thread_rng();
+
+    match input_size {
+        1 => "true".to_string(),
+        2 => "(some true) ".to_string(),
+        n => {
+            // assuming n is a multiple of 8
+            let mut val = String::new();
+            let mult = n / 8;
+            for _ in 0..mult {
+                let name = helper_generate_rand_char_string(5);
+                val.push_str(&*format!("({} {}) ", name, rng.gen::<u16>()));
+            }
+            format!("(tuple {}) ", val).to_string()
+        }
+    }
+}
 
 // generate arithmetic function call
-pub fn gen_arithmetic(function_name: &'static str, scale: u16, input_size: u16) -> String {
+pub fn gen_arithmetic(function_name: &'static str, scale: u16, input_size: u16) -> (Option<String>, String) {
```

Comment:
Thoughts on making the return type of the gens a named struct instead of an unnamed tuple?

Reply:
Discussed off of GH; will leave a comment explaining return type with the gen function